Lib/subprocess.py

```diff
@@ -386,6 +387,17 @@ def run(*popenargs, input=None, timeout=None, check=False, **kwargs):
     If timeout is given, and the process takes too long, a TimeoutExpired
     exception will be raised.
 
+    If the current process encounters any exception (e.g. KeyboardInterrupt) and
+    cleanup_timeout is non-zero, the child process is given some additional
+    time to finish before it is killed, to allow potential clean-up
+    operations in the child to complete. During this time, a second exception
+    will kill the child immediately.
+
+    If cleanup_timeout is not specified, then a default value is chosen:
+    If timeout was provided, then any remaining time from the original timeout
+    is used.  Otherwise, if the exception is a KeyboardInterrupt, the child is
+    given 1 second to clean up. Otherwise, the child is killed immediately.
```

Comment:
If cleanup timeout is not specified, there should be no cleanup timeout.

Reply:
>If cleanup timeout is not specified, there should be no cleanup timeout.

I assume you chose that phrasing to be pointedly simple, but, unfortunately, I think your phrasing is not as simple as it appears.  In the `subprocess` API and docs, "no timeout" (i.e. `timeout=None`) actually means "timeout=infinity".  That's probably not what you mean.

I think you meant to propose: "If cleanup timeout is not specified, there should be a cleanup timeout of 0.0".

Reply:
@stuarteberg I agree that "no cleanup timeout" is not clear enough, but it is not the same as cleanup timeout of 0 seconds.

How about:

If kill_timeout is not specified, the child process will be killed immediately when the timeout expire. If specified, the child process will be terminated gracefully using SIGTERM. If the process did not terminate within kill_timeout seconds, it will be killed.

This should be modified if graceful termination should be the default behavior. I'm not sure what should be the default, but this can be left for another PR.